src/v0/destinations/klaviyo/util.js

```diff
@@ -1,115 +1,74 @@
-const { httpGET } = require('../../../adapters/network');
-
+const { defaultRequestConfig } = require('rudder-transformer-cdk/build/utils');
 const { WhiteListedTraits } = require('../../../constants');
 
 const {
-  defaultRequestConfig,
   constructPayload,
   getFieldValueFromMessage,
-  isDefinedAndNotNull,
-  removeUndefinedValues,
   defaultPostRequestConfig,
   extractCustomFields,
   removeUndefinedAndNullValues,
+  defaultBatchRequestConfig,
 } = require('../../util');
 
-const { BASE_ENDPOINT, LIST_CONF, MAPPING_CONFIG, CONFIG_CATEGORIES } = require('./config');
-const {
-  getDynamicErrorType,
-  processAxiosResponse,
-} = require('../../../adapters/utils/networkUtils');
-const { NetworkError } = require('../../util/errorTypes');
-const tags = require('../../util/tags');
+const { BASE_ENDPOINT, MAPPING_CONFIG, CONFIG_CATEGORIES } = require('./config');
 
 /**
- * This function is used to check if the user/profile already exists or not, if already exists unique person_id
- * for that user is getting returned else false is returned.
- * Docs: https://developers.klaviyo.com/en/reference/get-profile-id
- * @param {*} message
- * @param {*} Config
- * @returns
+ * This function is used for creating response for subscribing users to a particular list.
+ * DOCS: https://www.klaviyo.com/docs/api/v2/lists
  */
-const isProfileExist = async (message, { Config }) => {
-  const { privateApiKey } = Config;
-  const userIdentifiers = {
+const subscribeUserToList = (message, traitsInfo, destination) => {
+  // listId from message properties are preferred over Config listId
+  const { privateApiKey, consent } = destination.Config;
+  let { listId } = destination.Config;
+  const targetUrl = `${BASE_ENDPOINT}/api/profile-subscription-bulk-create-jobs`;
+  const subscriptionObj = {
     email: getFieldValueFromMessage(message, 'email'),
-    external_id: getFieldValueFromMessage(message, 'userId'),
     phone_number: getFieldValueFromMessage(message, 'phone'),
   };
 
-  // eslint-disable-next-line no-restricted-syntax
-  for (const id in userIdentifiers) {
-    if (isDefinedAndNotNull(userIdentifiers[id])) {
-      // eslint-disable-next-line no-await-in-loop
-      const profileResponse = await httpGET(`${BASE_ENDPOINT}/api/v2/people/search`, {
-        header: {
-          Accept: 'application/json',
-        },
-        params: {
-          api_key: privateApiKey,
-          [id]: userIdentifiers[id],
-        },
-      });
-      const processedProfileResponse = processAxiosResponse(profileResponse);
-      if (processedProfileResponse.status === 200 && processedProfileResponse.response?.id) {
-        return processedProfileResponse.response.id;
+  if (traitsInfo?.properties?.consent || consent) {
+    const subscribeConsent = traitsInfo.properties.consent || consent;
```

Comment:
This can still cause an error like `Cannot find property consent of undefined`. Let's stick optional chaining here as well